src/test/java/org/jabref/logic/bibtex/comparator/BibStringDiffTest.java

```diff
@@ -0,0 +1,34 @@
+package org.jabref.logic.bibtex.comparator;
+
+import java.util.List;
+
+import org.jabref.model.database.BibDatabase;
+import org.jabref.model.entry.BibtexString;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BibStringDiffTest {
+
+    private final BibDatabase originalDataBase = mock(BibDatabase.class);
+    private final BibDatabase newDataBase = mock(BibDatabase.class);
+    private final BibStringDiff diff = new BibStringDiff(new BibtexString("name2", "content2"), new BibtexString("name2", "content3"));
+
+    @BeforeEach
+    void setUp() {
+        when(originalDataBase.hasNoStrings()).thenReturn(false);
+        when(newDataBase.hasNoStrings()).thenReturn(false);
+        when(originalDataBase.getStringValues()).thenReturn(List.of(new BibtexString("name", "content"), new BibtexString("name2", "content2")));
+        when(newDataBase.getStringValues()).thenReturn(List.of(new BibtexString("name", "content"), new BibtexString("name2", "content3")));
+    }
+
+    @Test
+    void compareTest() {
+        BibStringDiff result = BibStringDiff.compare(originalDataBase, newDataBase).get(0);
+        assertEquals(diff, result);
```

Comment:
Do not extract entries from list; test list directly

```suggestion
        List<BibStringDiff> result = BibStringDiff.compare(originalDataBase, newDataBase);
        assertEquals(List.of(diff), result);
```

Reply:
Thanks for the feedback, will definetly do that in the future.